common/lib/xmodule/xmodule/partitions/partitions.py

```diff
@@ -161,13 +171,16 @@ def from_json(value):
         if value["version"] == 1:
             # If no scheme was provided, set it to the default ('random')
             scheme_id = UserPartition.VERSION_1_SCHEME
-        elif value["version"] == UserPartition.VERSION:
+        elif value["version"] in [2, UserPartition.VERSION]:
             if "scheme" not in value:
                 raise TypeError("UserPartition dict {0} missing value key 'scheme'".format(value))
+
             scheme_id = value["scheme"]
         else:
             raise TypeError("UserPartition dict {0} has unexpected version".format(value))
```

Comment:
@wedaly @jimabramson when we will roll back from`UserPartition` with `VERSION = 3` to  `VERSION = 2` then course having `VerificationPartitionScheme` which uses `VERSION = 3` will start giving exception here due to which those courses won't load.

One possible solution would be to log the error here instead of raising exception and return `None`. Then in `from_json` method of `UserPartitionList` in file [inheritance.py](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/inheritance.py#L22) filter out the `None` objects.
```python
return filter(None, [UserPartition.from_json(v) for v in values])
```